lib/ansible/modules/storage/dellemc/dellemc_pmax_createsg.py

```diff
@@ -0,0 +1,200 @@
+      - "String up to 64 Characters no special character other than
+      underscore sets a label to make volumes easily identified on hosts can
+      run Dell EMC inq utility command to see this label is  inq -identifier
+      device_name"
+    required: false
+  user:
+    description:
+      - "Unisphere username"
+  password:
+    description:
+      - "password for Unisphere user"
+  workload:
+    description:
+      - "Block workload type, optional and can only be set on VMAX3 Hybrid
+      Storage Arrays. Default None."
+    required: false
+requirements:
+  - Ansible
+  - "Unisphere for PowerMax version 9.0 or higher."
+  - "VMAX All Flash, VMAX3, or PowerMax storage Array."
+  - "PyU4V version 3.0.0.8 or higher using PIP python -m pip install PyU4V"
+'''
+EXAMPLES = '''
+---
+- name: "Provision Storage For DB Cluster"
+  connection: local
+  hosts: localhost
+  vars:
+    array_id: 000197600156
+    password: smc
+    sgname: Ansible_SG
+    unispherehost: "192.168.1.123"
+    universion: "90"
+    user: smc
+    verifycert: false
+
+  tasks:
+    - name: "Create New Storage Group and add data volumes"
+      dellemc_pmax_createsg:
+        array_id: "{{array_id}}"
+        cap_unit: GB
+        num_vols: 1
+        password: "{{password}}"
+        sgname: "{{sgname}}"
+        slo: Diamond
+        srp_id: SRP_1
+        unispherehost: "{{unispherehost}}"
+        universion: "{{universion}}"
+        user: "{{user}}"
+        verifycert: "{{verifycert}}"
+        vol_size: 1
+        workload: None
+        volumeIdentifier: 'REDO'
+'''
+RETURN = '''
+'''
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    changed = False
+    module = AnsibleModule(
+        argument_spec=dict(
+            sgname=dict(type='str', required=True),
+            unispherehost=dict(required=True),
+            universion=dict(type='int', required=False),
+            verifycert=dict(type='bool', required=True),
+            user=dict(type='str', required=True),
+            password=dict(type='str', required=True, no_log=True),
+            array_id=dict(type='str', required=True),
+            srp_id=dict(type='str', required=False),
+            slo=dict(type='str', required=False),
+            workload=dict(type='str', required=False),
+            num_vols=dict(type='int', required=True),
+            vol_size=dict(type='int', required=True),
+            cap_unit=dict(type='str', required=True, choices=['GB',
+                                                              'TB',
+                                                              'MB', 'CYL']),
+            volumeIdentifier=dict(type='str', required=False)))
+    try:
+        import PyU4V
+    except:
+        module.fail_json(
+            msg='Requirements not met PyU4V is not installed, please install '
+                'via PIP')
+        module.exit_json(changed=changed)
+
+    conn = PyU4V.U4VConn(server_ip=module.params['unispherehost'], port=8443,
```

Comment:
If this is going to be common to other DellEMC modules then you can move this method to `module_utils/dellemc.py`.